harvard-lts/etd_dash_service
# Etd 160 jobmon

**Adding jobmon functionality.**
* * *

**JIRA Ticket**: [(link)](https://jira.huit.harvard.edu/browse/ETD-211)
(parent ticket: https://jira.huit.harvard.edu/browse/ETD-160)

# What does this Pull Request do?
Adds jobmon functionality. This pr slightly refactors existing util scripts and uses them in the worker.py to report job status to the jobmon. Also modifies message generated by scripts/invoke_tasks, and changes test to match.

# How should this be tested?

Local:
Checkout this branch locally
Start and exec into local docker (see readme)
run > pytest
Dev (all steps below done):
Deploy to dev as trial branch (done)
Find etd dash docker on dev[1-5] on dev and exec in
Run > python3 scripts/invoke_task.py
Tail the log, should complete with a "no files found in dropbox" message
Go to jobmon-dev, should see similar report, like this one (click Message Log button):
https://jobmon-dev.lib.harvard.edu/job_summary/job_id/37/run_id/1690380560#messagelog_1
rerun the invoke task script after copying over a submission packet from hegel archive to incoming, message should show success into etd dash, like this one (Click Message Log button):
https://jobmon-dev.lib.harvard.edu/job_summary/job_id/37/run_id/1690380698#messagelog_1


# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? Unit tests currently testing rewrite_mets, meeting to review coverage for legacy code upcoming
- integration tests? The integration test will be updated in the integration project

# Interested parties
Tag (@ mention) interested parties
